scipy/ndimage/_multimethods.py

```diff
@@ -0,0 +1,1065 @@
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def minimum(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def maximum(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def median(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def minimum_position(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def maximum_position(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def extrema(input, labels=None, index=None):
+    return input, labels, index
+
+
+@_create_ndimage(_input_labels_index_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def center_of_mass(input, labels=None, index=None):
+    return input, labels, index
+
+
+def _histogram_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for ``histogram``.
+    """
+    def self_method(input, min, max, bins, labels=None, index=None,
+                    *args, **kwargs):
+        return (dispatchables[0], min, max, bins,
+                dispatchables[1], dispatchables[2]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_histogram_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def histogram(input, min, max, bins, labels=None, index=None):
+    return input, labels, index
+
+
+def _watershed_ift_arg_replacer(args, kwargs, dispatchables):
+    """
+    uarray argument replacer for ``watershed_ift``.
+    """
+    def self_method(input, markers, structure=None, output=None,
+                    *args, **kwargs):
+        return (dispatchables[0], dispatchables[1],
+                dispatchables[2], dispatchables[3]) + args, kwargs
+
+    return self_method(*args, **kwargs)
+
+
+@_create_ndimage(_watershed_ift_arg_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def watershed_ift(input, markers, structure=None, output=None):
+    return input, markers, structure, _mark_non_coercible(output)
+
+
+############################################
+# morphology multimethods
+############################################
+
+
+@_create_ndimage(_input_replacer)
+@all_of_type(np.ndarray)
+@_get_docs
+def iterate_structure(structure, iterations, origin=None):
+    return (structure,)
+
+
+def _identity_arg_replacer(args, kwargs, arrays):
+    """
+    uarray argument replacer when dispatchables are empty.
+    """
+    return args, kwargs
+
+
+@_create_ndimage(_identity_arg_replacer)
+def generate_binary_structure(rank, connectivity):
+    return
```

Comment:
```suggestion
    return ()
```
needs to return a tuple, not None